zero_bin/rpc/src/main.rs

```diff
@@ -1,98 +1,201 @@
 use std::env;
 use std::sync::Arc;
 
+use alloy::primitives::B256;
+use alloy::providers::Provider;
 use alloy::rpc::types::eth::BlockId;
 use alloy::rpc::types::{BlockNumberOrTag, BlockTransactionsKind};
-use clap::{Parser, ValueHint};
+use alloy::transports::Transport;
+use anyhow::anyhow;
+use clap::{Args, Parser, Subcommand, ValueHint};
 use futures::StreamExt;
+use prover::BlockProverInput;
 use rpc::provider::CachedProvider;
-use rpc::{retry::build_http_retry_provider, RpcType};
+use rpc::{retry::build_http_retry_provider, RpcParams, RpcType};
 use tracing_subscriber::{prelude::*, EnvFilter};
 use url::Url;
 use zero_bin_common::pre_checks::check_previous_proof_and_checkpoint;
 use zero_bin_common::version;
 use zero_bin_common::{block_interval::BlockInterval, prover_state::persistence::CIRCUIT_VERSION};
 
-#[derive(Parser)]
-pub enum Cli {
-    /// Fetch and generate prover input from the RPC endpoint
+#[derive(Args, Clone, Debug)]
+pub(crate) struct RpcConfig {
```

Comment:
Now that I see this new struct, this is exactly what `RpcParams` on the leader side is, so would need unification

Reply:
Right, this was developed in parallel on `continuations` branch, hence the unintented duplication.

Reply:
Opened a new ticket for this https://github.com/0xPolygonZero/zk_evm/issues/548, worth thinking about what is the most optimal way to unify the parameters